sklearn/utils/validation.py

```diff
@@ -155,6 +156,38 @@ def _shape_repr(shape):
     return "(%s)" % joined
 
 
+def check_memory(memory):
+    """Check that the ``memory`` is Memory-like.
+
+    Memory-like means that ``memory`` can be converted into a
+    sklearn.externals.joblib.Memory instance (typically a str denoting the
+    ``cachedir``) or has the same interface.
+
+    Parameters
+    ----------
+    memory : Memory-like or None
+
+    Returns
+    -------
+    memory : object with the sklearn.externals.joblib.Memory interface
+
+    Raises
+    ------
+    ValueError
+        if ``memory`` is not Memory-like
+    """
+
+    if memory is None:
+        memory = Memory(cachedir=None, verbose=0)
+    elif isinstance(memory, six.string_types):
+        memory = Memory(cachedir=memory, verbose=0)
+    elif not hasattr(memory, 'cache'):
+        raise ValueError("'memory' is not a string or a Memory instance "
```

Comment:
Improve wording:
```py
"'memory' should be None, a string or have the same interface as sklearn.externals.joblib.Memory.
Got {} instead.
```

That means you need to amend the tests accordingly (if you change the error message).